### PR TITLE
LPS-102563 | Update SelectLayout component

### PR DESCRIPTION
**Pull request by markocikos**
Hey @wincent ,

This is a followup on https://github.com/wincent/liferay-portal/pull/98, now a complete solution.

Changes:
- Added styles to `Treeview`, to match current styles in portal.
- Applied styles to `layout-taglib` (`site-navigation-menu-item-layout`) and `asset-categories-selector-web`, basically removing all or nearly all existing styles
- Added small extension to `Treeview`, to handle extended state of nodes given as props. This is used in `layout-taglib`.
- Fixed a dependency setting

Please review the code! If all is ok, forward to @ealonso

Thank you!

### CODE REVIEW

**Comment by markocikos** [comment]
@wincent I added commit with cleanup of asset-categories-selector-web dependencies. With this, we can close https://issues.liferay.com/browse/LPS-102822

File: modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/treeview/Treeview.scss [hunk 1]
```diff
@@ -1,80 +1,150 @@
+$black: #272833;
+$blue: #80acff;
+$light-gray: #e7e7ed;
+$gray: #869cad;
+$white: #fff;
```
Comment (wincent) [inline modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/treeview/Treeview.scss]: Please sort.

**Comment by wincent** [comment]
ci:test:sf

**Comment by wincent** [comment]
ci:test:stable

**Comment by wincent** [comment]
ci:test:relevant

**Comment by wincent** [comment]
Ok, after `ant all` I was able to actually test the effect of this on the asset categories selector — it seems we've lost the focused styling which is important for you to be able to see where you are when using keyboard navigation.

Master (blue background is a selected node, blue ring is a focused node):

master

This PR (node is focused but there is no ring):
 
pr

Also a heads up to @sandrodw3, these changes might affect your use of the treeview too. What's the status of that right now?

**Comment by sandrodw3** [comment]
Hey @wincent, @markocikos. Yes, my use of the treeview is already in master. I finally overwrote some styles as it was only a few lines of CSS. I suppose that if classnames are still the same It shouldn't affect me. And I agree with you about the focused styling, as it's important for the keyboard nav!

**Comment by markocikos** [comment]
@wincent 

>  we've lost the focused styling which is important for you to be able to see where you are when using keyboard navigation.

The nodes do not focus properly in the `Treeview` component. We intercept focusing event and end up only setting `.focused` css class. Is this intentional behavior, or is it a bug and we want to both focus and set the class? I can change styles to rely on `.focused` instead of `:focus`, but that seems bad for accessibility. What is the reason for setting the `.focused` class?

**Comment by wincent** [comment]
The *intent* is that we actually focus the nodes; see [this line in the `useFocus()` hook](https://github.com/liferay/liferay-portal/blob/dae37cf57738575076e2d9a358608b2c177fda2b/modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/treeview/useFocus.js#L28) which calls `focus()` on the DOM element. The `NodeList` component [uses `useFocus()`](https://github.com/liferay/liferay-portal/blob/3dc759cac3e370ca6c32ce93dbbea52d876f25da/modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/treeview/NodeList.js#L35) and [`NodeListItem` does the same](https://github.com/liferay/liferay-portal/blob/3dc759cac3e370ca6c32ce93dbbea52d876f25da/modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/treeview/NodeListItem.js#L28) (both are `div` elements with `tabIndex` set for keyboard navigation and accessibility. AFAIK that's all working.

If you adjust the styles to target `:focus` instead of `focused` things *might* work, but I suspect other things may break. IIRC correctly, the reason I used a `.focused` class instead of just targeting with `:focus` is that I was having some trouble fighting against Clay's styles, so I wanted to ever-so-slightly separate the DOM notion of focus from the UI notion of how we draw it: that's why you can see [in the `TreeviewCard` component](https://github.com/liferay/liferay-portal/blob/3dc759cac3e370ca6c32ce93dbbea52d876f25da/modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/treeview/TreeviewCard.js#L55) which is responsible for actually rendering the stuff that the user sees, that's where we make use of the `.focused` class.

**Comment by markocikos** [comment]
@wincent Found the bug; I'll be sending a fix as a part of the followup PR. It's on [this line](https://github.com/liferay/liferay-portal/blob/3dc759cac3e370ca6c32ce93dbbea52d876f25da/modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/treeview/Treeview.js#L604), should be 'FOCUS' and not 'ACTIVATE'.

**Comment by wincent** [comment]
Are you sure about that? The purpose of that `handleFocus` and `handleBlur` is to track when the focus enters or leaves the `Treeview`. When it enters, it becomes active and starts managing focus internally (hence the `ACTIVATE` action); when it leaves it stops managing focus (hence `DEACTIVATE`). The `FOCUS` action is for focusing individual lists/items within the treeview but I see that right now we're not explicitly firing that from it anywhere; I think that is a (different) bug.

**Comment by markocikos** [comment]
Yeah, it's not that simple. With the change, focus is working but I need to click twice to select, only can select previously focused. Also, we are setting `tabindex=-1` on all nodes, so keyboard navigation is not working. Was that intentional?

**Comment by wincent** [comment]
> Was that intentional?

Yes, the `Treeview` component wants to manage keyboard navigation, so it sets the tabindex to -1 so that it can be managed programmatically, following this bit in the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex):

> A *negative value* (usually `tabindex="-1"`) means that the element is not reachable via sequential keyboard navigation, but could be focused with Javascript or visually. It's mostly useful to create accessible widgets with JavaScript.

**Pull request closed without merging**